common/rust/cctrusted_base/src/eventlog.rs

```diff
@@ -291,14 +405,61 @@ impl TcgEventLog {
         index += event_size as usize;
 
         Ok((
-            TcgImrEvent {
+            TcgEventLog {
+                rec_num,
                 imr_index,
                 event_type,
                 digests,
                 event_size,
                 event,
+                extra_info: HashMap::new(),
             },
             index.try_into().unwrap(),
         ))
     }
+
+    /***
+       Parse ascii IMA events gathered during runtime.
+
+       Sample event and format:
+       IMR index | Template hash | Template name | Event data according to template
+       10 1e762ca412a3ef388ddcab416e2eb382d9d1e356 ima-ng sha384:74ccc46104f42db070375e6876a23aeaa3c2ae458888475baaa171c3fb7001b0fc385ed08420d5f60620924fc64d0b80 /etc/lsb-release
+
+       Args:
+           event: IMA ascii raw event
+
+       Returns:
+           A TcgEventLog object containing the ima event log
+    */
+    fn parse_ima_event_log(&mut self, data: &str) -> Result<TcgEventLog, anyhow::Error> {
+        let elements: Vec<&str> = data.trim_matches(' ').split(' ').collect();
+
```

Comment:
One minor comment, add comments for the index numbers, like elements[0] contains imr index, elements[1] contains xxx, ...

Reply:
updated